### PR TITLE
Implement new functions and few improvements for supervisor

### PR DESCRIPTION
**Pull request by nghiant2710**

### CODE REVIEW

**Comment by dfunckt** [comment]
Code/Docs-wise LGTM, but let's wait for @pcarranzav to confirm the API calls to device.

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -525,7 +543,7 @@ u'python-sdk-test-device'
 ```
 ### Function: restart(uuid)
 
-Restart a device. This function only works if you log in using credentials or Auth Token.
+Restart an user application container on device. This function only works if you log in using credentials or Auth Token.
```
Comment (pcarranzav) [inline DOCUMENTATION.md]: Nit: `a` instead of `an`

File: resin/models/supervisor.py [hunk 2]
```diff
@@ -310,7 +333,242 @@ def restart(self, app_id, device_id=None):
...
+            )
+
+    def stop_application(self, app_id=None, device_uuid=None):
+        """
+        Temporarily stops an user application container. Application container will not be removed after invoking this function and a reboot or supervisor restart will cause the container to start again.
+        This function requires supervisor v1.8 or higher.
+        No need to set device_uuid and app_id if command is sent to the API on device.
+
+        Args:
+            app_id (Optional[str]): application id.
```
Comment (pcarranzav) [inline resin/models/supervisor.py]: `app_id` should be required, not optional, in these and the methods below

File: DOCUMENTATION.md (hunk 1 above)
Comment (nghiant2710) [inline DOCUMENTATION.md]: Updated!

File: resin/models/supervisor.py (hunk 2 above)
Comment (nghiant2710) [inline resin/models/supervisor.py]: Thanks for pointing this out, @pcarranzav 
Updated!

**Comment by pcarranzav** [comment]
LGTM

**Pull request merged**